toolkit/tools/internal/safechroot/safechroot.go

```diff
@@ -441,50 +451,85 @@ func cleanupAllChroots() {
 	inChrootMutex.Lock()
 
 	// mount is only supported in regular pipeline
+	failedToUnmount := false
 	if buildpipeline.IsRegularBuild() {
 		// Cleanup chroots in LIFO order incase any are interdependent (e.g. nested safe chroots)
 		logger.Log.Info("Cleaning up all active chroots")
 		for i := len(activeChroots) - 1; i >= 0; i-- {
 			logger.Log.Infof("Cleaning up chroot (%s)", activeChroots[i].rootDir)
-			err := activeChroots[i].unmountAndRemove(leaveChrootOnDisk)
+			err := activeChroots[i].unmountAndRemove(leaveChrootOnDisk, isFinalAttempt)
 			// Perform best effort cleanup: unmount as many chroots as possible,
 			// even if one fails.
 			if err != nil {
 				logger.Log.Errorf("Failed to unmount chroot (%s)", activeChroots[i].rootDir)
+				failedToUnmount = true
 			}
 		}
 	}
 
-	logger.Log.Info("Cleanup finished")
+	if failedToUnmount {
+		logger.Log.Fatalf("Failed to unmount a chroot, manual unmount required")
+	} else {
+		logger.Log.Info("Cleanup finished")
+	}
 }
 
 // unmountAndRemove retries to unmount directories that were mounted into
 // the chroot until the unmounts succeed or too many failed attempts.
 // This is to avoid leaving folders like /dev mounted when the chroot folder is forcefully deleted in cleanup.
 // Iff all mounts were successfully unmounted, the chroot's root directory will be removed if requested.
-func (c *Chroot) unmountAndRemove(leaveOnDisk bool) (err error) {
+func (c *Chroot) unmountAndRemove(leaveOnDisk, isFinalAttempt bool) (err error) {
 	const (
-		totalAttempts = 3
-		retryDuration = time.Second
-		unmountFlags  = 0
+		retryDuration       = time.Second
+		totalAttemptsNormal = 3
+		totalAttemptsFinal  = 5
```

Comment:
Why do you need retries on a lazy detach?

Reply:
Paranoia mostly, but I've not seen it fail on the 1st attempt... given that maybe I'll just simplify it to use the same retry logic as the normal attempt with the understanding that it is expected to work the 1st time.

Reply:
Does it matter if the lazy detach succeeds? Isn't it just a mitigation against a bad situation?

Reply:
As long as the chroot leaves the system in a state where the tools can be run again, I think it counts as "success". I'd rather we be able to explicitly unmount, but practice has shown that just doesn't always happen. Our `make clean` has used lazy unmounts for a long time via `safeunmount.sh`, and I'm not aware of any negative affects from using the lazy flag.